File: Java Based Archives/ViewModelwithLiveData/app/src/main/java/com/android/viewmodel/MainActivityViewModel.java
package com.android.viewmodel;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class MainActivityViewModel extends ViewModel {
    private int clickCount = 0;
    private MutableLiveData<Integer> countLiveData = new MutableLiveData<>();

    // Method to retrieve initial count
    public MutableLiveData<Integer> getInitialCount() {
        // Setting initial count value to LiveData
        countLiveData.setValue(clickCount);
        return countLiveData;
    }

    // Method to increment count
    public void getCurrentCount() {
        // Incrementing count and updating LiveData
        clickCount += 1;
        countLiveData.setValue(clickCount);
    }
}

File: Views/LiveDataFundamentals/app/src/main/java/com/example/viewmodeldemo2/MainActivityViewModel.kt
package com.example.viewmodeldemo2

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class MainActivityViewModel(startingTotal: Int) : ViewModel() {
    private var total = MutableLiveData<Int>()
    val totalData: LiveData<Int>
        get() = total

    init {
        total.value = startingTotal
    }
    fun set2Total(input: Int?) {
        total.value = (total.value ?: 0) + (input ?: 0)
    }

}
